src/devtools/client/debugger/src/client/index.ts

```diff
@@ -32,19 +33,27 @@ type $FixTypeLater = any;
 
 async function setupDebugger() {
   const sourceInfos: $FixTypeLater[] = [];
-  // @ts-expect-error `sourceMapURL` doesn't exist?
-  await ThreadFront.findSources(({ sourceId, url, sourceMapURL }) =>
+  await ThreadFront.findSources(newSource => {
+    // @ts-expect-error `sourceMapURL` doesn't exist?
+    const { sourceId, url, sourceMapURL } = newSource;
     sourceInfos.push({
       type: "generated",
       data: prepareSourcePayload({
         actor: sourceId,
         url,
         sourceMapURL,
       }),
-    })
-  );
+    });
+
+    // Can also probably just shove these into an array right here and dispatch
+    // them all at the end.
+    store.dispatch(addSource(newSource));
+  });
   await store.dispatch(actions.newQueuedSources(sourceInfos));
   store.dispatch({ type: "SOURCES_LOADED" });
+
+  // This fires off the coercing of sources into a more workable form
+  store.dispatch(allSourcesReceived());
```

Comment:
:hand: We could key off the existing `"SOURCES_LOADED"` action instead

Reply:
I went slightly the other way and changed the place that was listening for that action to listen for `sources/allSourcesReceived` instead, but same effect 👍